testing/test_pickle_checkpoint_step.py

```diff
@@ -5,55 +5,125 @@
 from py._path.local import LocalPath
 
 from neuraxle.checkpoints import PickleCheckpointStep
+from neuraxle.hyperparams.space import HyperparameterSamples
 from neuraxle.pipeline import Pipeline
 from neuraxle.steps.util import TapeCallbackFunction, TransformCallbackStep
 
-data_inputs = np.ones((1, 1))
-expected_outputs = np.ones((2, 2))
+EXPECTED_TAPE_AFTER_CHECKPOINT = ["2", "3"]
 
+data_inputs = np.array([1, 2])
+expected_outputs = np.array([2, 3])
+expected_rehashed_data_inputs = ['44f9d6dd8b6ccae571ca04525c3eaffa', '898a67b2f5eeae6393ca4b3162ba8e3d']
 
-def test_should_save_checkpoint_pickle(tmpdir: LocalPath):
-    tape = TapeCallbackFunction()
-    pickle_checkpoint_step = PickleCheckpointStep('1', tmpdir)
+
+def create_pipeline_without_hyperparams(pickle_checkpoint_step, tape):
+    pipeline = Pipeline(
+        steps=[
+            ('a', TransformCallbackStep(tape.callback, ["1"])),
+            ('pickle_checkpoint', pickle_checkpoint_step),
+            ('c', TransformCallbackStep(tape.callback, ["2"])),
+            ('d', TransformCallbackStep(tape.callback, ["3"]))
+        ]
+    )
+    return pipeline
+
+
+def create_pipeline_with_hyperparams(pickle_checkpoint_step, tape):
```

Comment:
This method could perhaps be merged with the above with `hyperparams=None` as a method argument.